xml/cl.xml

```diff
@@ -5484,6 +5524,7 @@ server's OpenCL/api-docs repository.
         <extension name="cl_khr_fp64" supported="opencl">
             <require>
                 <type name="CL/cl.h"/>
+                <type name="cl_double"/>
```

Comment:
This is debatably the right thing to do, but `cl_double` is currently defined unconditionally in `cl_platform.h`.  This means that the extension headers should never emit anything for `cl_double`, even if it matches what is currently in `cl_platform.h`, at least for C99:

https://stackoverflow.com/questions/26240370/why-are-typedef-identifiers-allowed-to-be-declared-multiple-times

So, what should we do about this?  Some options:

1. Consider `cl_double` to effectively be part of the API, unconditionally, meaning it's part of "OpenCL 1.0".  This feels a little dirty, but I can't find any headers without `cl_double` so I think it's encoding reality.
2. Consider `cl_double` to be an extension type only pre-OpenCL 1.2, and a core type for OpenCL 1.2 and newer.  This is probably the "most correct", though I'm worried what this will break.
3. Keep `cl_double` defined as-is in the headers and more-or-less as-defined here in the XML file, but add some sort of conditions or special-case code in the extension header generation scripts to avoid emitting anything for it.

(1) is the easiest thing to do.  I wouldn't recommend (2).  I think we could make (3) work but we'd need to figure out exactly what we want to do.